src/Elastic.Markdown/Helpers/Preloader.cs

```diff
@@ -0,0 +1,37 @@
+// Licensed to Elasticsearch B.V under one or more agreements.
+// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information
+
+using System.Diagnostics;
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace Elastic.Markdown.Helpers;
+
+public static partial class FontPreloader
+{
+	private static List<string>? FontUriCache = null!;
```

Comment:
readonly field or getter only property. 

Also IReadOnlyCollection instead of List.

Reply:
Does this need to be thread safe?

Reply:
Ah. Nevermind. If I understood correctly, the whole logic is only executed once.